pyomo/core/base/constraint.py

```diff
@@ -109,14 +110,14 @@ def C_rule(model, i, j):
 
     model.c = ConstraintList(expr=simple_constraintlist_rule(...))
     """
-    return rule_wrapper(
-        rule,
-        {
-            None: ConstraintList.End,
-            True: Constraint.Feasible,
-            False: Constraint.Infeasible,
-        },
-    )
+    result_map = {None: ConstraintList.End}
+    for l_type in native_logical_types:
+        result_map[l_type(True)] = Constraint.Feasible
+        result_map[l_type(False)] = Constraint.Infeasible
+    # Note: some logical types has the same as bool (e.g., np.bool_), so
```

Comment:
I think you meant this?
```suggestion
    # Note: some logical types act the same as bool (e.g., np.bool_), so
```